internal/reconciliations/gateway/virtual_service.go

```diff
@@ -0,0 +1,70 @@
+package gateway
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"github.com/kyma-project/api-gateway/apis/operator/v1alpha1"
+	"github.com/kyma-project/api-gateway/internal/reconciliations"
+	"istio.io/client-go/pkg/apis/networking/v1beta1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	kymaGatewayVirtualServiceName      = "kyma-gateway"
+	kymaGatewayVirtualServiceNamespace = "istio-system"
+)
+
+//go:embed virtual_service.yaml
+var virtualServiceManifest []byte
+
+func reconcileKymaGatewayVirtualService(ctx context.Context, k8sClient client.Client, apiGatewayCR v1alpha1.APIGateway, domain string) error {
+	name := kymaGatewayVirtualServiceName
```

Comment:
We already have this local constants, why do we use again some variables for name and namespace? We can just use the constants.